plugins/pulp_rpm/plugins/distributors/yum/metadata/updateinfo.py

```diff
@@ -102,6 +164,16 @@ def add_unit_metadata(self, item):
                                       'epoch': package['epoch'] or '0',
                                       'arch': package['arch'],
                                       'src': package.get('src', '') or ''}
+
+                if repo_unit_nevra is not None:
```

Comment:
I think this is a reasonable approach given the way this function is setup. But just to say it, in an ideal world it would be better if there were a way to move this logic outside the function. Then ``add_unit_metadata`` could focus on adding metadata, and not on deciding which data to add. The obvious design would be to ``filter`` or similar up on line 159, but because of the transformation that happens in between there and here, that isn't convenient. without changing lots of stuff.

Oh well. Just an observation.